lib/surface.ex

```diff
@@ -181,9 +181,14 @@ defmodule Surface do
       |> Keyword.merge(static_props)
       |> rename_id_if_stateless(module.component_type())
 
+    renamed_slot_props =
+      module
+      |> mapping_to_rename_slots_with_as_option()
```

Comment:
And here also. Maybe rename it to something like `map_slots_to_assigns`. The important thing is that it should always be clear if we're dealing with slots (external) or assigns (internal) when naming variables or functions. There should be places where we don't follow this rules, especially regarding props. In the near future we'll probably need to have option `:as` for properties too as they also represent external inputs. So since we we'll have two possible different sets of names around (slots or assigns), I think is a good idea to start following this golden rule.

Reply:
Naming is hard and you are definitely better than me :smile: I updated the code accordingly to your comments.

I started to add the `:as` option for properties but after few minutes I forgot the usefulness of doing this. Why would we like to have something like this?

```elixir
defmodule Component

  prop title, :string, as: :default_title

  render(assigns) do
    ~H"""
       <div>
          {{ @default_tilte }}
       </div>  
    """
  end
end
```

Adding a `:as` option to prop will from my point of view solve the same issue we had with `slot`. As we already offer a proper way to handle this, I am not sure we should offer a second way of fixing the same issue. It may be confusing and add complexity. For example, the user could write something like that:
```elixir
defmodule Component

  slot title, as: :default_title
  prop title, :string, as: :default_title

  render(assigns) do
    ~H"""
       <div>
          <slot name="title" />
          {{ @default_tilte }}
       </div>  
    """
  end
end
```

Reply:
@Malian you're right. Maybe we won't need `:as` for `prop` anymore. Let's move on without it and if it turns out to be useful in the future, we can discuss again.